packages/with-env/src/index.ts

```diff
@@ -1,11 +1,19 @@
-export default function withEnv<T = any>(env: string, callback: () => T): T {
-  const previousEnv = process.env.NODE_ENV;
+import {Env} from './types';
+
+export default function withEnv<T = any>(env: Env, callback: () => T): T {
+  const envToSet = typeof env === 'string' ? {NODE_ENV: env} : env;
+
+  const previousEnv = Object.keys(envToSet).reduce((previousEnv, key) => {
+    previousEnv[key] = process.env[key];
+    return previousEnv;
+  }, {});
+
+  Object.assign(process.env, envToSet);
+
   function resetEnv() {
-    process.env.NODE_ENV = previousEnv;
+    Object.assign(process.env, previousEnv);
```

Comment:
one thing to note is that this won't un-assign anything that was previously undefined, so we may want to instead clear off all the keys from `process.env` before doing this assign.

Reply:
Great catch, thank you!
I've fixed it, added a test for that case and force-pushed the commit (as I only rebased against master :))